packages/@react-aria/overlays/src/calculatePosition.ts

```diff
@@ -149,17 +152,24 @@ function getDelta(
   // is portaled somewhere other than the body and has an ancestor with
   // position: relative/absolute, it will be different.
   containerDimensions: Dimensions,
-  padding: number
+  padding: number,
+  containerOffsetWithBoundary: Offset
 ) {
   let containerScroll = containerDimensions.scroll[axis];
-  let boundaryHeight = boundaryDimensions[AXIS_SIZE[axis]];
-  let startEdgeOffset = offset - padding - containerScroll;
-  let endEdgeOffset = offset + padding - containerScroll + size;
-
-  if (startEdgeOffset < 0) {
-    return -startEdgeOffset;
-  } else if (endEdgeOffset > boundaryHeight) {
-    return Math.max(boundaryHeight - endEdgeOffset, -startEdgeOffset);
+  // The height/width of the boundary. Matches the axis along which we are adjusting the overlay position
+  let boundarySize = boundaryDimensions[AXIS_SIZE[axis]];
+  // Calculate the edges of the boundary (accomodating for the boundary padding) and the edges of the overlay.
+  let boundaryStartEdge = boundaryDimensions.scroll[AXIS[axis]] + padding;
+  let boundaryEndEdge = boundarySize + boundaryDimensions.scroll[AXIS[axis]] - padding;
+  let startEdgeOffset = offset - containerScroll + containerOffsetWithBoundary[axis];
+  let endEdgeOffset = offset - containerScroll + size + containerOffsetWithBoundary[axis];
```

Comment:
In the case where the menu is portalled to a container that isn't the body, we need to adjust the offset calculated to fit the same coordinate system that the boundary uses hence the addition of the containerOffsetWIthBounday[axis] here. This fixes the getDelta for submenus since they are positioned with respect to their parent menu, not the page body.

Also moved the padding accommodation to boundary edge calculations since it felt more intuitive to put it there

Reply:
This is potentially a case for introducing a new prop. The container for the submenu isn't actually the body but a div within its parent menu that the submenu gets portalled into. Said container matches the width of the viewport but doesn't actually have a height and so we apply a container padding of 0 for the submenu. In this case we could consider adding a `boundaryPadding` prop to `useOverlayPosition`/`calculatePosition` and all the other hooks that call them. 

In many places in `calculatePosition`, `padding` seems to be equivalent to `boundaryPadding` already, but `useOverlayPosition` passes `containerPadding` as `padding` since we most likely considered container === boundary?

Reply:
I'll handle this as a followup, will write a issue to track